subprojects/publish/src/main/java/org/gradle/api/publish/plugins/PublishingPlugin.java

```diff
@@ -66,7 +66,7 @@ public void apply(final Project project) {
         PublicationContainer publications = instantiator.newInstance(DefaultPublicationContainer.class, instantiator);
 
         // TODO Registering an extension should register it with the model registry as well
-        project.getExtensions().create(PublishingExtension.NAME, DefaultPublishingExtension.class, repositories, publications);
+        project.getExtensions().create(PublishingExtension.NAME, PublishingExtension.class, DefaultPublishingExtension.class, repositories, publications);
```

Comment:
Strictly speaking, these are breaking changes, as (i think) you can no longer ask for an extension of type `DefaultPublishingExtension` in build logic. Which is fine, I think, but we should mention in the release notes.